TestFoundation/TestNSDate.swift

```diff
@@ -41,6 +42,13 @@ class TestNSDate : XCTestCase {
         let d = NSDate()
         XCTAssertNotNil(d)
     }
+
+    func test_DescriptionWithLocale() {
+        let d = NSDate(timeIntervalSince1970: 0)
+        XCTAssertEqual(d.descriptionWithLocale(NSLocale(localeIdentifier: "ja_JP")), "1970年1月1日木曜日 0時00分00秒 GMT")
+        XCTAssertEqual(d.descriptionWithLocale(NSLocale(localeIdentifier: "de_DE")), "Donnerstag, 1. Januar 1970 um 00:00:00 GMT")
+        XCTAssertEqual(d.descriptionWithLocale(NSLocale(localeIdentifier: "ru_RO")), "четверг, 1 января 1970 г., 0:00:00 GMT")
+    }
```

Comment:
Let's add a test for a nil locale as well. Obviously the result depends on user locale, so just verify that it returns a non-empty string.